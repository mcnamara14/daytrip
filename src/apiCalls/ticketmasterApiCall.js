import { ticketmasterApiKey } from './apiKeys/ticketmasterApiKey';

export const ticketmasterApiCallRecentEvents = async (city, state, timeNow, timeIn2Days) => {
  const url = `https://app.ticketmaster.com/discovery/v2/events.json?startDateTime=${timeNow}&endDateTime=${timeIn2Days}&sort=date,name,asc&city=${city}&stateCode=${state}&apikey=${ticketmasterApiKey}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;

    return events
  } catch (error) {
      alert(error)
  }
}





