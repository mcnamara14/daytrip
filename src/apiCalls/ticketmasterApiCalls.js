import { ticketmasterApiKey } from './apiKeys/ticketmasterApiKey';
import { 
  cleanRecentEventsSearch, 
  cleanRecentEvents 
} from '../dataCleaners/index';

export const fetchRecentEvents = async (city, state, timeNow) => {
  const prefix = 'https://app.ticketmaster.com/discovery/v2/events.json?';
  const start = `startDateTime=${timeNow}`;
  const sort = 'sort=date,name,asc';
  const userCity = `city=${city}`;
  const userState = `stateCode=${state}`;
  const apiKey = `apikey=${ticketmasterApiKey}`;

  const url = `${prefix}&${start}&${sort}&${userCity}&${userState}&${apiKey}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;
    console.log(events)

    return events;
  } catch (error) {
    alert(error);
  }
};

export const fetchRecentEventsOnSearch = async (
  userCity, 
  userState, 
  timeNow, 
  keyword) => {
  const prefix = 'https://app.ticketmaster.com/discovery/v2/events.json?';
  const start = `startDateTime=${timeNow}`;
  const sort = 'sort=date,name,asc';
  const city = `city=${userCity}`;
  const state = `stateCode=${userState}`;
  const input = `keyword=${keyword}`;
  const key = `apikey=${ticketmasterApiKey}`;

  const url = `${prefix}${start}&${sort}&${city}&${state}&${input}&${key}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;
  
    return cleanRecentEventsSearch(events);
  } catch (error) {
    console.log(error)
  }

};

export const fetchSelectedEvent = async (id) => {
  const prefix = `https://app.ticketmaster.com/discovery/v2/events/${id}.json?`;
  const url = `${prefix}apikey=${ticketmasterApiKey}`;

  try {
    const response = await fetch(url);
    const event = await response.json();
    
    return cleanRecentEvents([event]);
  } catch (error) {
    console.log('error');
  }
};










