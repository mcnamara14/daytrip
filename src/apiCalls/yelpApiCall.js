import { yelpApiKey } from './apiKeys/yelpApiKey';
import * as cleaners from '../dataCleaners/index';
import { suggestedRestaurantsCleaner } from '../dataCleaners/suggestedRestaurantsCleaner';

export const yelpApiCall = async () => {
  let headers = new Headers();
  headers.append("Authorization", "Bearer " + yelpApiKey);

  const result = await fetch("https://cors-anywhere.herokuapp.com/https://api.yelp.com/v3/businesses/search?term=delis&latitude=37.786882&longitude=-122.399972", {
    headers
  });
  const data = await result.json();
};

export const yelpFetchRestaurants = async (latitude, longitude, price, category) => {
  let headers = new Headers();
  headers.append("Authorization", "Bearer " + yelpApiKey);
  
  const result = await fetch(`https://cors-anywhere.herokuapp.com/https://api.yelp.com/v3/businesses/search?categories=${category}&latitude=${latitude}&longitude=${longitude}&price=${price}&radius=2500&sort_by=rating`, {
    headers
  });
  const data = await result.json();
  const restaurants = data.businesses;

  return suggestedRestaurantsCleaner(restaurants)
};