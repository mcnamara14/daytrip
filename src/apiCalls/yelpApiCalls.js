import { yelpApiKey } from './apiKeys/yelpApiKey';
import { 
  suggestedRestaurantsCleaner 
} from '../dataCleaners/suggestedRestaurantsCleaner';

export const fetchRestaurantsAndBars = async (
  latitude, 
  longitude, 
  price, 
  category) => {

  const corsAnywhereUrl = 'https://cors-anywhere.herokuapp.com/'
  const prefix = `${corsAnywhereUrl}https://api.yelp.com/v3/businesses/search?`;
  const cat = `categories=${category}`;
  const lat = `latitude=${latitude}`;
  const long = `longitude=${longitude}`;
  const cost = `price=${price}`;
  const radius = `radius=2500`;
  const sort = 'sort_by=rating';

  let headers = new Headers();
  headers.append("Authorization", "Bearer " + yelpApiKey);
  
  const result = 
    await fetch(`${prefix}${cat}&${lat}&${long}&${cost}&${radius}&${sort}`, 
      {
        headers
      });
  const data = await result.json();
  const restaurants = data.businesses;
  
  return suggestedRestaurantsCleaner(restaurants);
};