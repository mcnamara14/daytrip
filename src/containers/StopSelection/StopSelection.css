.stopsSelectionContainer h3 {
  color: #fff;
  font-weight: 300;
  text-transform: capitalize;
  text-align: left;
  font-size: 1.4rem;
  margin-top: 0px;
}

.stopsSelection button {
  margin-top: 30px;
  box-shadow: 1px 1px 1px rgba(116,87,117,1);
  border: none;
  border-radius: 3px;
  padding: 15px 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  color: #fff;
  font-size: 1rem;
  background: linear-gradient(45deg, rgba(213,144,159,1) 0%, rgba(157,108,128,1) 100%);
}

.stopsSelectionContainer button:hover {
  background: rgba(217,73,44,1);
}

.beforeEvent, .afterEvent {
  width: 40%;
}

.filterTitle {
  color: #ddd;
  font-weight: 300;
  text-transform: capitalize;
}

.priceRange {
  border-radius: 3px;
  display: flex;
  height: 45px;
  align-items: center;
}

.priceRange span {
  margin-right: 5px;
  padding: 13px 18.5px;
  color: #fff;
  border: 1px solid #fff;
}

.priceRange span.selected {
  background: #e58da6;
}

.priceRange span:hover {
  cursor: pointer;
  background: #e58da6;
}

.priceRange span.last {
  margin-right: 0;
}

.Select-menu-outer div div:hover {
  background: #eef5ff;
  cursor: pointer;
}

.Select-menu-outer div div{
  background: #eef5ff;
  cursor: pointer;
  background: #fff;
}

.Select-menu-outer div {
  background: #f8f8f8;
  cursor: pointer;
}

.Select-menu-outer div {
  border: 1px solid #eee;
  padding: 10px;
}

.Select-menu-outer .is-selected {
  background: #fff;
}

.Select-menu-outer .is-selected:hover {
  background: #fff;
  cursor: pointer;
}

.Select-menu  {
}