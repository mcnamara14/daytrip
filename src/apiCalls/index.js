import { ticketmasterApiCall } from './ticketmasterApiCall'; 
import { yelpApiCall } from './yelpApiCall'; 

export {
  ticketmasterApiCall,
  yelpApiCall
};