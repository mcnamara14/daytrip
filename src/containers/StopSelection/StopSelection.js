import React, { Component } from 'react';
import './StopSelection.css';
import { beforeEventCategoryCleaner } from '../../dataCleaners/beforeEventCategoryCleaner';
import { afterEventCategoryCleaner } from '../../dataCleaners/afterEventCategoryCleaner';
import { Async } from 'react-select';

export class StopSelection extends Component {
  constructor() {
    super();

    this.state = {
      selectedOption: {}
    }
  };

beforeEventCategories = async () => {
  return beforeEventCategoryCleaner();
}

afterEventCategories = async () => {
  return afterEventCategoryCleaner();
}

onSelect = (selectedOption) => {
  this.setState({ selectedOption });
  
  if (selectedOption) {
    console.log(selectedOption.label)
  }
}

render() {
  const { selectedOption} = this.state;
  const beforeAfter = this.props.type === 'before' ? 'before' : 'after';
  const restaurantBar = this.props.type === 'before' ? 'restaurant' : 'bar';
  const beforeAfterCategories = this.props.type === 'before' ? this.beforeEventCategories : this.afterEventCategories;
  const className = beforeAfter + 'Event';

  return (
    <div className={className}>
      <h3>{beforeAfter} the event</h3>
      <p className="filterTitle">{restaurantBar} category</p>
      <Async
        name="searchEventsInput"
        loadOptions={beforeAfterCategories}
        value={selectedOption}
        onChange={this.onSelect}
        placeholder="Choose a category"
      />
      <p className="filterTitle">Price</p>
      <div className="priceRange">
        <span>$</span>
        <span>$$</span>
        <span>$$$</span>
        <span className="last">$$$$</span>
      </div>
      <button>Submit</button>
    </div>
  );
}
}
