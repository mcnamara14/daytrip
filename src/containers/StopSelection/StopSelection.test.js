import React from 'react';
import ReactDOM from 'react-dom';
import { StopSelection } from './StopSelection';
import { shallow } from 'enzyme';
import { 
  mockCleanRestaurantAndBar, 
  mockCleanRecentEvents,
  mockCleanSuggestedRestaurant
 } from '../../mockData';
import * as fetchCalls from '../../apiCalls';
jest.mock('../../apiCalls/yelpApiCalls');

describe('StopSelection', () => {
it('should match the snapshot', () => {
  const wrapper = shallow(<StopSelection />)

  expect(wrapper).toMatchSnapshot();
});

  describe('onSelect', () => {
    it('should set state to selected option', () => {
      const wrapper = shallow(<StopSelection />)

      wrapper.instance().onSelect(mockCleanRestaurantAndBar)
     
      expect(wrapper.state('selectedOption')).toEqual(mockCleanRestaurantAndBar)
    });
  })

  describe('changePriceRange', () => {
    it('should add price range to state when its not included already', () => {
      const wrapper = shallow(<StopSelection />);
      wrapper.setState({
        priceRanges: ['2', '5']
      })

      wrapper.instance().changePriceRange('1');

      const expected = ['2', '5', '1'];
     
      expect(wrapper.state('priceRanges')).toEqual(expected)
    });

    it('should remove price range from state when its included already', () => {
      const wrapper = shallow(<StopSelection />);
      wrapper.setState({
        priceRanges: ['2', '5']
      })

      wrapper.instance().changePriceRange('2');

      const expected = ['5'];
     
      expect(wrapper.state('priceRanges')).toEqual(expected)
    });
  })

  describe('toggleEventError', () => {
    it('should call toggleEventError with an arg of true then call again with an arg of false after 2 seconds', () => {
      const wrapper = shallow(<StopSelection toggleEventError={jest.fn()} />);
      jest.useFakeTimers();

      wrapper.instance().toggleEventError();

      expect(wrapper.instance().props.toggleEventError).toHaveBeenCalledWith(true)

      jest.runAllTimers();

      expect(wrapper.instance().props.toggleEventError).toHaveBeenCalledWith(false)
    })
  })

  describe('storeRestaurantsOrBars', () => {
    it('should call fetchRestaurantsAndBars with the correct args when there is a selectedEvent and selected restaurant or bar category', async () => {
      const mockSelectedEvent = mockCleanRecentEvents.events[0];

      const wrapper = shallow(<StopSelection toggleEventError={jest.fn()} selectedEvent={mockSelectedEvent} storeSuggestedRestaurants={jest.fn()} storeSuggestedBars={jest.fn()} />);
      
      wrapper.setState({
        selectedOption: {
          alias: "newamerican",
          label: "New American"
        }
      })
      
      await wrapper.instance().storeRestaurantsOrBars();
      
      const expected = ["39.735001", "-105.020401", "", "newamerican"];

      expect(fetchCalls.fetchRestaurantsAndBars).toHaveBeenCalledWith(...expected);
    });

    it('should call storeSuggestedRestaurants with the correct args when the type is before', async () => {
      const wrapper = shallow(<StopSelection toggleEventError={jest.fn()} selectedEvent={{title: 'T-Swift'}} storeSuggestedRestaurants={jest.fn()} storeSuggestedBars={jest.fn()} type={'before'} />);
      fetchCalls.fetchRestaurantsAndBars.mockImplementation(() => mockCleanSuggestedRestaurant)

      wrapper.setState({
        selectedOption: {title: 'The Matchbox'}
      })

      await wrapper.instance().storeRestaurantsOrBars();

      const expected = mockCleanSuggestedRestaurant
      
      expect(wrapper.instance().props.storeSuggestedRestaurants).toHaveBeenCalledWith(expected);
    });

    it('should call storeSuggestedBars with the correct args when the type is after', async () => {
      const wrapper = shallow(<StopSelection toggleEventError={jest.fn()} selectedEvent={{title: 'T-Swift'}} storeSuggestedRestaurants={jest.fn()} storeSuggestedBars={jest.fn()} type={'after'} />);
      fetchCalls.fetchRestaurantsAndBars.mockImplementation(() => mockCleanSuggestedRestaurant)

      wrapper.setState({
        selectedOption: {title: 'The Matchbox'}
      })

      await wrapper.instance().storeRestaurantsOrBars();

      const expected = mockCleanSuggestedRestaurant
      
      expect(wrapper.instance().props.storeSuggestedBars).toHaveBeenCalledWith(expected);
    });

  });
})
