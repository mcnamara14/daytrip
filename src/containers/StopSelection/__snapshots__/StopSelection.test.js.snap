// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopSelection should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StopSelection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          after
           the event
</h3>,
        <p
          className="filterTitle"
>
          bar
           category
</p>,
        <Select
          arrowRenderer={[Function]}
          autosize={true}
          backspaceRemoves={true}
          backspaceToRemoveMessage="Press backspace to remove {label}"
          clearAllText="Clear all"
          clearRenderer={[Function]}
          clearValueText="Clear value"
          clearable={true}
          closeOnSelect={true}
          deleteRemoves={true}
          delimiter=","
          disabled={false}
          escapeClearsValue={true}
          filterOptions={[Function]}
          ignoreAccents={true}
          ignoreCase={true}
          inputProps={Object {}}
          isLoading={false}
          joinValues={false}
          labelKey="label"
          matchPos="any"
          matchProp="any"
          menuBuffer={0}
          menuRenderer={[Function]}
          multi={false}
          name="searchEventsInput"
          noResultsText="No results found"
          onBlurResetsInput={true}
          onChange={[Function]}
          onCloseResetsInput={true}
          onSelectResetsInput={true}
          openOnClick={true}
          optionComponent={[Function]}
          options={
                    Array [
                              Object {
                                "alias": "airportlounges",
                                "label": "Airport Lounges",
                              },
                              Object {
                                "alias": "beerbar",
                                "label": "Beer Bar",
                              },
                              Object {
                                "alias": "champagne_bars",
                                "label": "Champagne Bars",
                              },
                              Object {
                                "alias": "cigarbars",
                                "label": "Cigar Bars",
                              },
                              Object {
                                "alias": "cocktailbars",
                                "label": "Cocktail Bars",
                              },
                            ]
          }
          pageSize={5}
          placeholder="Choose a category"
          removeSelected={true}
          required={false}
          rtl={false}
          scrollMenuIntoView={true}
          searchable={true}
          simpleValue={false}
          tabSelectsValue={true}
          trimFilter={true}
          value={null}
          valueComponent={[Function]}
          valueKey="value"
/>,
        <p
          className="filterTitle"
>
          Price
</p>,
        <div
          className="priceRange"
>
          <span
                    className="priceOne"
                    onClick={[Function]}
          >
                    $
          </span>
          <span
                    className="priceTwo"
                    onClick={[Function]}
          >
                    $$
          </span>
          <span
                    className="priceThree"
                    onClick={[Function]}
          >
                    $$$
          </span>
          <span
                    className="priceFour last"
                    onClick={[Function]}
          >
                    $$$$
          </span>
</div>,
      ],
      "className": "afterEvent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "after",
            " the event",
          ],
        },
        "ref": null,
        "rendered": Array [
          "after",
          " the event",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "bar",
            " category",
          ],
          "className": "filterTitle",
        },
        "ref": null,
        "rendered": Array [
          "bar",
          " category",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowRenderer": [Function],
          "autosize": true,
          "backspaceRemoves": true,
          "backspaceToRemoveMessage": "Press backspace to remove {label}",
          "clearAllText": "Clear all",
          "clearRenderer": [Function],
          "clearValueText": "Clear value",
          "clearable": true,
          "closeOnSelect": true,
          "deleteRemoves": true,
          "delimiter": ",",
          "disabled": false,
          "escapeClearsValue": true,
          "filterOptions": [Function],
          "ignoreAccents": true,
          "ignoreCase": true,
          "inputProps": Object {},
          "isLoading": false,
          "joinValues": false,
          "labelKey": "label",
          "matchPos": "any",
          "matchProp": "any",
          "menuBuffer": 0,
          "menuRenderer": [Function],
          "multi": false,
          "name": "searchEventsInput",
          "noResultsText": "No results found",
          "onBlurResetsInput": true,
          "onChange": [Function],
          "onCloseResetsInput": true,
          "onSelectResetsInput": true,
          "openOnClick": true,
          "optionComponent": [Function],
          "options": Array [
            Object {
              "alias": "airportlounges",
              "label": "Airport Lounges",
            },
            Object {
              "alias": "beerbar",
              "label": "Beer Bar",
            },
            Object {
              "alias": "champagne_bars",
              "label": "Champagne Bars",
            },
            Object {
              "alias": "cigarbars",
              "label": "Cigar Bars",
            },
            Object {
              "alias": "cocktailbars",
              "label": "Cocktail Bars",
            },
          ],
          "pageSize": 5,
          "placeholder": "Choose a category",
          "removeSelected": true,
          "required": false,
          "rtl": false,
          "scrollMenuIntoView": true,
          "searchable": true,
          "simpleValue": false,
          "tabSelectsValue": true,
          "trimFilter": true,
          "value": null,
          "valueComponent": [Function],
          "valueKey": "value",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Price",
          "className": "filterTitle",
        },
        "ref": null,
        "rendered": "Price",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="priceOne"
              onClick={[Function]}
>
              $
</span>,
            <span
              className="priceTwo"
              onClick={[Function]}
>
              $$
</span>,
            <span
              className="priceThree"
              onClick={[Function]}
>
              $$$
</span>,
            <span
              className="priceFour last"
              onClick={[Function]}
>
              $$$$
</span>,
          ],
          "className": "priceRange",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$",
              "className": "priceOne",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "$",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$$",
              "className": "priceTwo",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "$$",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$$$",
              "className": "priceThree",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "$$$",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$$$$",
              "className": "priceFour last",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "$$$$",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            after
             the event
</h3>,
          <p
            className="filterTitle"
>
            bar
             category
</p>,
          <Select
            arrowRenderer={[Function]}
            autosize={true}
            backspaceRemoves={true}
            backspaceToRemoveMessage="Press backspace to remove {label}"
            clearAllText="Clear all"
            clearRenderer={[Function]}
            clearValueText="Clear value"
            clearable={true}
            closeOnSelect={true}
            deleteRemoves={true}
            delimiter=","
            disabled={false}
            escapeClearsValue={true}
            filterOptions={[Function]}
            ignoreAccents={true}
            ignoreCase={true}
            inputProps={Object {}}
            isLoading={false}
            joinValues={false}
            labelKey="label"
            matchPos="any"
            matchProp="any"
            menuBuffer={0}
            menuRenderer={[Function]}
            multi={false}
            name="searchEventsInput"
            noResultsText="No results found"
            onBlurResetsInput={true}
            onChange={[Function]}
            onCloseResetsInput={true}
            onSelectResetsInput={true}
            openOnClick={true}
            optionComponent={[Function]}
            options={
                        Array [
                                    Object {
                                      "alias": "airportlounges",
                                      "label": "Airport Lounges",
                                    },
                                    Object {
                                      "alias": "beerbar",
                                      "label": "Beer Bar",
                                    },
                                    Object {
                                      "alias": "champagne_bars",
                                      "label": "Champagne Bars",
                                    },
                                    Object {
                                      "alias": "cigarbars",
                                      "label": "Cigar Bars",
                                    },
                                    Object {
                                      "alias": "cocktailbars",
                                      "label": "Cocktail Bars",
                                    },
                                  ]
            }
            pageSize={5}
            placeholder="Choose a category"
            removeSelected={true}
            required={false}
            rtl={false}
            scrollMenuIntoView={true}
            searchable={true}
            simpleValue={false}
            tabSelectsValue={true}
            trimFilter={true}
            value={null}
            valueComponent={[Function]}
            valueKey="value"
/>,
          <p
            className="filterTitle"
>
            Price
</p>,
          <div
            className="priceRange"
>
            <span
                        className="priceOne"
                        onClick={[Function]}
            >
                        $
            </span>
            <span
                        className="priceTwo"
                        onClick={[Function]}
            >
                        $$
            </span>
            <span
                        className="priceThree"
                        onClick={[Function]}
            >
                        $$$
            </span>
            <span
                        className="priceFour last"
                        onClick={[Function]}
            >
                        $$$$
            </span>
</div>,
        ],
        "className": "afterEvent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "after",
              " the event",
            ],
          },
          "ref": null,
          "rendered": Array [
            "after",
            " the event",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "bar",
              " category",
            ],
            "className": "filterTitle",
          },
          "ref": null,
          "rendered": Array [
            "bar",
            " category",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrowRenderer": [Function],
            "autosize": true,
            "backspaceRemoves": true,
            "backspaceToRemoveMessage": "Press backspace to remove {label}",
            "clearAllText": "Clear all",
            "clearRenderer": [Function],
            "clearValueText": "Clear value",
            "clearable": true,
            "closeOnSelect": true,
            "deleteRemoves": true,
            "delimiter": ",",
            "disabled": false,
            "escapeClearsValue": true,
            "filterOptions": [Function],
            "ignoreAccents": true,
            "ignoreCase": true,
            "inputProps": Object {},
            "isLoading": false,
            "joinValues": false,
            "labelKey": "label",
            "matchPos": "any",
            "matchProp": "any",
            "menuBuffer": 0,
            "menuRenderer": [Function],
            "multi": false,
            "name": "searchEventsInput",
            "noResultsText": "No results found",
            "onBlurResetsInput": true,
            "onChange": [Function],
            "onCloseResetsInput": true,
            "onSelectResetsInput": true,
            "openOnClick": true,
            "optionComponent": [Function],
            "options": Array [
              Object {
                "alias": "airportlounges",
                "label": "Airport Lounges",
              },
              Object {
                "alias": "beerbar",
                "label": "Beer Bar",
              },
              Object {
                "alias": "champagne_bars",
                "label": "Champagne Bars",
              },
              Object {
                "alias": "cigarbars",
                "label": "Cigar Bars",
              },
              Object {
                "alias": "cocktailbars",
                "label": "Cocktail Bars",
              },
            ],
            "pageSize": 5,
            "placeholder": "Choose a category",
            "removeSelected": true,
            "required": false,
            "rtl": false,
            "scrollMenuIntoView": true,
            "searchable": true,
            "simpleValue": false,
            "tabSelectsValue": true,
            "trimFilter": true,
            "value": null,
            "valueComponent": [Function],
            "valueKey": "value",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Price",
            "className": "filterTitle",
          },
          "ref": null,
          "rendered": "Price",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="priceOne"
                onClick={[Function]}
>
                $
</span>,
              <span
                className="priceTwo"
                onClick={[Function]}
>
                $$
</span>,
              <span
                className="priceThree"
                onClick={[Function]}
>
                $$$
</span>,
              <span
                className="priceFour last"
                onClick={[Function]}
>
                $$$$
</span>,
            ],
            "className": "priceRange",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$",
                "className": "priceOne",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "$",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$$",
                "className": "priceTwo",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "$$",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$$$",
                "className": "priceThree",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "$$$",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$$$$",
                "className": "priceFour last",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "$$$$",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
